tests/fulltextindent.js

```diff
@@ -0,0 +1,124 @@
+					doc = model.document;
+				} );
+		} );
+
+		it( 'should keep margin in paragraph', () => {
+			editor.setData( '<p style="margin:24px 8px 12px 6px;">foo</p>' );
+
+			const paragraph = doc.getRoot().getChild( 0 );
+
+			expect( paragraph.hasAttribute( 'margin' ) ).to.be.true;
+			expect( paragraph.getAttribute( 'margin' ) ).to.deep.equal( [ 24, 8, 12, 6 ] );
+
+			expect( getViewData( editor.editing.view, { withoutSelection: true } ) )
+				.to.equal( '<p style="margin:24px 8px 12px 6px">foo</p>' );
+		} );
+
+		it( 'should keep margin with only 3 values in paragraph', () => {
+			editor.setData( '<p style="margin:24px 8px 12px;">foo</p>' );
+
+			const paragraph = doc.getRoot().getChild( 0 );
+
+			expect( paragraph.hasAttribute( 'margin' ) ).to.be.true;
+			expect( paragraph.getAttribute( 'margin' ) ).to.deep.equal( [ 24, 8, 12 ] );
+
+			expect( getViewData( editor.editing.view, { withoutSelection: true } ) )
+				.to.equal( '<p style="margin:24px 8px 12px">foo</p>' );
+		} );
+
+		it( 'should keep margin with only 2 values in paragraph', () => {
+			editor.setData( '<p style="margin:24px 8px;">foo</p>' );
+
+			const paragraph = doc.getRoot().getChild( 0 );
+
+			expect( paragraph.hasAttribute( 'margin' ) ).to.be.true;
+			expect( paragraph.getAttribute( 'margin' ) ).to.deep.equal( [ 24, 8 ] );
+
+			expect( getViewData( editor.editing.view, { withoutSelection: true } ) )
+				.to.equal( '<p style="margin:24px 8px">foo</p>' );
+		} );
+
+		it( 'should keep margin with only 1 value in paragraph', () => {
+			editor.setData( '<p style="margin:24px;">foo</p>' );
+
+			const paragraph = doc.getRoot().getChild( 0 );
+
+			expect( paragraph.hasAttribute( 'margin' ) ).to.be.true;
+			expect( paragraph.getAttribute( 'margin' ) ).to.deep.equal( [ 24 ] );
+
+			expect( getViewData( editor.editing.view, { withoutSelection: true } ) )
+				.to.equal( '<p style="margin:24px">foo</p>' );
+		} );
+
+		it( 'should keep negative margin in paragraph', () => {
+			editor.setData( '<p style="margin:-24px -8px -12px -6px;">foo</p>' );
+
+			const paragraph = doc.getRoot().getChild( 0 );
+
+			expect( paragraph.hasAttribute( 'margin' ) ).to.be.true;
+			expect( paragraph.getAttribute( 'margin' ) ).to.deep.equal( [ -24, -8, -12, -6 ] );
+
+			expect( getViewData( editor.editing.view, { withoutSelection: true } ) )
+				.to.equal( '<p style="margin:-24px -8px -12px -6px">foo</p>' );
+		} );
+
+		it( 'should convert pt in px and keep margin in paragraph', () => {
+			editor.setData( '<p style="margin:24pt 8pt 12pt 6pt;">foo</p>' );
+
+			const paragraph = doc.getRoot().getChild( 0 );
+			const convertedValue = getRoundedValues( '24pt 8pt 12pt 6pt' );
+
+			expect( paragraph.hasAttribute( 'margin' ) ).to.be.true;
+			expect( paragraph.getAttribute( 'margin' ) ).to.deep.equal( convertedValue );
+
+			expect( getViewData( editor.editing.view, { withoutSelection: true } ) )
+				.to.equal( '<p style="margin:32px 11px 16px 8px">foo</p>' );
```

Comment:
I think you should rather do something like in https://github.com/concordnow/ckeditor5-tabulation-converter/blob/master/tests/paragraptabulationconverter.js right ?

```
.to.equal( `<p style="margin:${ getRoundedValue('24pt') }px ${ getRoundedValue('8pt') }px ${ getRoundedValue('12pt') }px ${ getRoundedValue('6pt') }px"` )
```

Reply:
Or rather use the rounded values from `convertedValues`

Reply:
Good remark I will do it now

Reply:
Done